### ISSUE TITLE
pvacseq installed but gives error message -bash: pvacseq: command not found

### ISSUE THREAD

**Issue by strixeh** [griffithlab/pVAC-Seq#94]
Hi,

I am a noob when it comes to programming, but I think I have installed pvacseq properly. Pvacseq will not run from the terminal (Im on OS X 10.11.4). 

>pvacseq 
-bash: pvacseq: command not found

>pvacseq install_vep_plugin
-bash: pvacseq: command not found

I have tried to update pvacseq and it says that requirements are already up to date
>pip3 install --upgrade pvacseq
Requirement already up-to-date: pvacseq in /Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages
Requirement already up-to-date: PyVCF in /Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages (from pvacseq)
Requirement already up-to-date: setuptools in /Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages (from PyVCF->pvacseq)

I have also tried to uninstall pvacseq, PyVCF, setuptools, update pip3 to 8.1.2, and then reinstall them all again, one by one. 

I thought it might be something wrong with PATH but I am not skilled enough to determine that
>echo $PATH
/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:/usr/X11/bin

I have followed the VEP tutorial and i can call VEP without problems. 

Any ideas?

Thanks,
Nils

**Comment by susannasiebert** [comment]
Hi Nils,

I'm sorry you're having problems with getting pvacseq to work. When you run `pip3 list` do you see pvacseq in the list? Also, what version do you see when running `python -V`?

**Comment by susannasiebert** [comment]
Also, do you have python 3.5 installed or are you using conda to emulate a python 3.5 environment?

**Comment by strixeh** [comment]
Hi - thanks for such a quick response!

>pip3 list
pip (8.1.2)
pvacseq (2.0.1)
PyVCF (0.6.8)
setuptools (24.0.2)

I installed python3.5:
>python3 -V
Python 3.5.2

I also have another python version installed, that comes with the OS X
>python -V
Python 2.7.10

**Comment by susannasiebert** [comment]
Ok, I think we're probably encountering some conflicts between python and python3 where python is the default but we're expecting python3 to be the default. What does `pip list` show you? Are you able to install pvacseq using `pip install pvacseq`?

**Comment by susannasiebert** [comment]
What we would recommend is that you use conda do emulate a python 3.5 environment. Conda only emulates one python version at a time so there are no conflicts. See [this link](http://conda.pydata.org/docs/download.html) for installing conda and [this one](http://conda.pydata.org/docs/py2or3.html) for setting up your environment. I presonally have been using Miniconda and have been finding it pretty straightforward to use once it is installed.

**Comment by strixeh** [comment]
Hi Susanna,

Again, thanks for a prompt reply!

So, this might be weird, but...

>pip list
-bash: pip: command not found

I guess this should be the case?

I will try using Conda, but would appreciate your input on the above still :).

**Comment by susannasiebert** [comment]
That is... weird. Try running `sudo apt-get install python-pip` and then try running `pip install pvacseq`.

**Comment by strixeh** [comment]
>sudo apt-get install python-pip 
sudo: apt-get: command not found

However, 

>sudo easy_install pip
Password:
Searching for pip
Reading https://pypi.python.org/simple/pip/
Best match: pip 8.1.2
Downloading https://pypi.python.org/packages/e7/a8/7556133689add8d1a54c0b14aeff0acb03c64707ce100ecd53934da1aa13/pip-8.1.2.tar.gz#md5=87083c0b9867963b29f7aba3613e8f4a
Processing pip-8.1.2.tar.gz
Writing /tmp/easy_install-xDiy32/pip-8.1.2/setup.cfg
Running pip-8.1.2/setup.py -q bdist_egg --dist-dir /tmp/easy_install-xDiy32/pip-8.1.2/egg-dist-tmp-8BLJ_k
warning: no previously-included files found matching '.coveragerc'
warning: no previously-included files found matching '.mailmap'
warning: no previously-included files found matching '.travis.yml'
warning: no previously-included files found matching '.landscape.yml'
warning: no previously-included files found matching 'pip/_vendor/Makefile'
warning: no previously-included files found matching 'tox.ini'
warning: no previously-included files found matching 'dev-requirements.txt'
warning: no previously-included files found matching 'appveyor.yml'
no previously-included directories found matching '.github'
no previously-included directories found matching '.travis'
no previously-included directories found matching 'docs/_build'
no previously-included directories found matching 'contrib'
no previously-included directories found matching 'tasks'
no previously-included directories found matching 'tests'
Adding pip 8.1.2 to easy-install.pth file
Installing pip script to /usr/local/bin
Installing pip2.7 script to /usr/local/bin
Installing pip2 script to /usr/local/bin

Installed /Library/Python/2.7/site-packages/pip-8.1.2-py2.7.egg
Processing dependencies for pip
Finished processing dependencies for pip
MAC186533:variant_effect_predictor nilsrudqvist$ pip list
altgraph (0.10.2)
bdist-mpkg (0.5.0)
bonjour-py (0.3)
macholib (1.5.1)
matplotlib (1.3.1)
modulegraph (0.10.4)
numpy (1.8.0rc1)
pip (8.1.2)
py2app (0.7.3)
pyobjc-core (2.5.1)
pyobjc-framework-Accounts (2.5.1)
pyobjc-framework-AddressBook (2.5.1)
pyobjc-framework-AppleScriptKit (2.5.1)
pyobjc-framework-AppleScriptObjC (2.5.1)
pyobjc-framework-Automator (2.5.1)
pyobjc-framework-CFNetwork (2.5.1)
pyobjc-framework-Cocoa (2.5.1)
pyobjc-framework-Collaboration (2.5.1)
pyobjc-framework-CoreData (2.5.1)
pyobjc-framework-CoreLocation (2.5.1)
pyobjc-framework-CoreText (2.5.1)
pyobjc-framework-DictionaryServices (2.5.1)
pyobjc-framework-EventKit (2.5.1)
pyobjc-framework-ExceptionHandling (2.5.1)
pyobjc-framework-FSEvents (2.5.1)
pyobjc-framework-InputMethodKit (2.5.1)
pyobjc-framework-InstallerPlugins (2.5.1)
pyobjc-framework-InstantMessage (2.5.1)
pyobjc-framework-LatentSemanticMapping (2.5.1)
pyobjc-framework-LaunchServices (2.5.1)
pyobjc-framework-Message (2.5.1)
pyobjc-framework-OpenDirectory (2.5.1)
pyobjc-framework-PreferencePanes (2.5.1)
pyobjc-framework-PubSub (2.5.1)
pyobjc-framework-QTKit (2.5.1)
pyobjc-framework-Quartz (2.5.1)
pyobjc-framework-ScreenSaver (2.5.1)
pyobjc-framework-ScriptingBridge (2.5.1)
pyobjc-framework-SearchKit (2.5.1)
pyobjc-framework-ServiceManagement (2.5.1)
pyobjc-framework-Social (2.5.1)
pyobjc-framework-SyncServices (2.5.1)
pyobjc-framework-SystemConfiguration (2.5.1)
pyobjc-framework-WebKit (2.5.1)
pyOpenSSL (0.13.1)
pyparsing (2.0.1)
python-dateutil (1.5)
pytz (2013.7)
scipy (0.13.0b1)
setuptools (1.1.6)
six (1.4.1)
xattr (0.6.4)
zope.interface (4.1.1)

Although

>MAC186533:variant_effect_predictor nilsrudqvist$ pip install pvacseq
Collecting pvacseq
  Could not find a version that satisfies the requirement pvacseq (from versions: )
No matching distribution found for pvacseq

**Comment by agraubert** [comment]
I'm curious if `pip3` will work for you.  Try `pip3 install pvacseq`

**Comment by susannasiebert** [comment]
Ah yeah, you're on Mac. I accidentally gave you the instructions for Linux. 

So it looks like pip is linked to python which is linked to python 2.7 and pip3 is linked to python3 which  is linked to python 3.5. However, we expect pip and python to point to python 3.5. Unfortunately, this default can't be changed easily and has the potential to break a lot of things which is why we recommend using conda.

We had a user with a similar problem (https://github.com/griffithlab/pVAC-Seq/issues/79) and he was able to get it to work by running `pip3 install --upgrade pip` twice which caused his pip to point to pip3 (for some strange reason). He was then able to run `pip install pvacseq` and get it to work. However, he also changed his symlinks and such so this might not work for you but it's worth a shot I guess.

**Comment by strixeh** [comment]
@agraubert `pip3 install pvacseq`has always worked, however, when I try to call pvacseq later, It returns `pvacseq: command not found`

@susannasiebert, thanks for the suggestion. I actually tried running `pip3 install --upgrade pip` twice but it does not help me. Any suggestion to why pip cant find any matching distributions found for pvacseq?

Thanks all!

**Comment by susannasiebert** [comment]
Because pip points to python 2.7 under the hood and since pvacseq is written in python 3.5 pip doesn't meet the requirements.

**Comment by strixeh** [comment]
@susannasiebert ah ok, and when I call `pvacseq`, my system believes it should use python 2.7 to call it? (i apologize for any wrong use of terms...)

**Comment by susannasiebert** [comment]
yeah, I believe that is what is going on.

**Comment by strixeh** [comment]
@susannasiebert ok, thanks. I will try using Anaconda instead. Should I try to uninstall anything on my computer before starting using Anaconda?

**Comment by susannasiebert** [comment]
You probably don't need to uninstall anything. When you setup your environment with conda it basically gives you a clean slate with whatever python version you specify.

**Comment by strixeh** [comment]
@susannasiebert Ok, thanks for all the help! I really appreciate it!

**Comment by strixeh** [comment]
@susannasiebert Was able to install pvacseq without any problems using Conda. Thanks for the great help and suggestion.

**Comment by susannasiebert** [comment]
Awesome! Glad to hear. I'll resolve this issue but feel free to put in a new issue should you run into any problems in the future.
